### PR TITLE
Training documentation update

### PR DESCRIPTION
**Pull request by laserkelvin**
This PR adds additional documentation pertaining to training, partially addressing #280:

- Task API details, up to date to what tasks available today
- Updated best practices, detailing target normalization and loss scaling

### CODE REVIEW

File: docs/source/best-practices.rst [hunk 1]
```diff
@@ -181,6 +181,81 @@ the accelerator.
...
+
+   Task(
+       ...,
+       normalize_kwargs={
+         "energy_mean": 0.0,
+         "energy_std": 1.0,
+   }
+   )
+
+The example above will normalize ``energy`` labelsm and can be substituted with
```
Comment (melo-gonzo) [inline docs/source/best-practices.rst:200]: Typo

File: docs/source/best-practices.rst [hunk 2]
```diff
@@ -223,6 +298,20 @@ inspired by observations made in LLM training research, where the breakdown of
 assumptions in the convergent properties of ``Adam``-like optimizers causes large
 spikes in the training loss. This callback can help identify these occurrences.
 
+The ``devset``/``fast_dev_run`` approach detailed above is also useful for testing
+engineering/infrastructure (e.g. accelerator offload and logging), but not necessarily
```
Comment (melo-gonzo) [inline docs/source/best-practices.rst:302]: fast_dev_run disables logging i believe.

**Review by melo-gonzo — APPROVED** [review]
Just a few comments on this one. Looks good! Feel free to merge when updated.

File: docs/source/best-practices.rst (hunk 2 above)
Comment (laserkelvin) [reply docs/source/best-practices.rst]: Good catch - fixed in [17d7582](https://github.com/IntelLabs/matsciml/pull/298/commits/17d75825146bd8ebb9966184a80d0604a58ba2de)

File: docs/source/best-practices.rst (hunk 1 above)
Comment (laserkelvin) [reply docs/source/best-practices.rst]: Fixed in [9231641](https://github.com/IntelLabs/matsciml/pull/298/commits/923164122cee903ace01d17db2d28f2b2b11d8c3)

**Pull request merged**